pkg/cvo/metrics.go

```diff
@@ -83,6 +106,10 @@ version for 'cluster', or empty for 'initial'.
 			Name: "cluster_installer",
 			Help: "Reports info about the installation process and, if applicable, the install tool.",
 		}, []string{"type", "version", "invoker"}),
+		clusterVersionOperatorSuccessfulUpdateRetrieval: prometheus.NewGaugeVec(prometheus.GaugeOpts{
+			Name: "cluster_version_operator_successful_update_retrieval",
+			Help: "Reports when updates were last succesfully retrieved and, if currently not able to retrieve updtaes, reports reason.",
+		}, []string{"lastTime", "reason", "message"}),
```

Comment:
`message` would be super-helpful, but... I'm concerned about label cardinality (apparently Prometheus doesn't like it if we have labels that have many different values, and `message` is basically unbounded).  We should probably stick to `lastTime` and `reason`, and have the alert message say "For more information..." and then... somehow point people at the condition.  By linking to a console view?  Or by saying "... see ClusterVersion's status.conditions"?  Or something...

Reply:
- How about if we just tell them to look in the cvo log and search for 'Reason'? It at least is an easy approach.
- Assume by "console view" you mean pop up a window of some kind with the message? Do you know if this approach used anywhere else OpenShift?
- I'll post a question to forum-monitoring for any recommendations/ideas as well.

Reply:
> Assume by "console view" you mean pop up a window of some kind with the message? Do you know if this approach used anywhere else OpenShift?

Yeah, I mean Administration > Cluster Settings > Overview, like [here][1].

[1]: https://docs.openshift.com/container-platform/4.2/updating/updating-cluster-between-minor.html#update-upgrading-web_updating-cluster-between-minor

Reply:
The base issue is that an unbounded label set, such as "labels.message"s strings, can cause an unbounded set of independent metrics rather than one continuous metric. I saw this behavior with the lastTime value when I wasn't careful about only setting it when it truly needed to be set. But do we really have this issue with 'message'? Our metric is only unique and alive for the period since LastSyncOrConfigChange and true, if the LastSyncOrConfigChange and Reason remain the same but the Message changes, which I think would be rare, we would get a new metric which would be incorrect. But as I said, that would be very rare.
Another approach is to go back to the way I had it with a separate "metric" just for the Alert.

Reply:
Following up on the console URI, the installer spits out:

https://console-openshift-console.apps.ci-ln-cy2dir2-d5d6b.origin-ci-int-aws.dev.rhcloud.com/

We want to link people to:

https://console-openshift-console.apps.ci-ln-cy2dir2-d5d6b.origin-ci-int-aws.dev.rhcloud.com/settings/cluster/

(so console URI + `/settings/cluster/`), where they will see a UI like:

![Screenshot from 2020-05-20 09-40-02](https://user-images.githubusercontent.com/209920/82473102-ef9f3780-9a7d-11ea-90c6-2f66a358ebc0.png)

Reply:
ACK. Is there a need for the get console URI method outside of the cvo package, e.g. should I create in lib perhaps?

Reply:
Done with 06c550f.